tools/vendor/vendor-check.py

```
@@ -0,0 +1,590 @@
+        for entry in component.get("files", []):
+            if not isinstance(entry, dict):
+                failures.add("C-FILES", "%s: stance is track but files[] carries "
+                                        "a bare path (%r) with no digest"
+                             % (cid, entry))
+                continue
+            # Non-None: the equality against `on_disk` above already proved every
+            # declared path is a real disk name.
+            rel = entry.get("path")
+            recorded = entry.get("sha256", "")
+            if not SHA256_RE.match(recorded):
+                failures.add("C-FILES", "%s: %s has no valid sha256" % (cid, rel))
```

Ruff (0.16.1)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)